tools/readahead.py

```diff
@@ -95,15 +95,18 @@
 """
 
 b = BPF(text=program)
-b.attach_kprobe(event="__do_page_cache_readahead", fn_name="entry__do_page_cache_readahead")
-b.attach_kretprobe(event="__do_page_cache_readahead", fn_name="exit__do_page_cache_readahead")
+ra_event = "__do_page_cache_readahead"
+if kernel_version_ge(5, 10):
+    ra_event = "do_page_cache_ra"
+b.attach_kprobe(event=ra_event, fn_name="entry__do_page_cache_readahead")
+b.attach_kretprobe(event=ra_event, fn_name="exit__do_page_cache_readahead")
```

Comment:
Let us just do first try attach_kprobe with __do_page_cache_readahead(), if failed, try do_page_cache_ra().  You can look at nfsdist.py as an example.

Reply:
Ok,  adjutsted.